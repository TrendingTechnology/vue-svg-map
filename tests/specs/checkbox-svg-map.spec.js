import Vue from 'vue'
import { mount } from '@vue/test-utils'
import map from '../mocks/fake-map'
import SvgMap from '../../src/components/svg-map'
import CheckboxSvgMap from '../../src/components/checkbox-svg-map'

describe('CheckboxSvgMap component', () => {
	const value = ['id0']
	let wrapper
	let svgMap
	let locations

	beforeEach(() => {
		wrapper = mount(CheckboxSvgMap, {
			propsData: { map, value },
		})
		svgMap = wrapper.find(SvgMap)
		locations = svgMap.findAll('path')
	})

	it('selects locations of inital value property', () => {
		expect(locations.at(0).attributes('aria-checked')).toBeTruthy()
		expect(locations.at(1).attributes('aria-checked')).toBeFalsy()
		expect(locations.at(2).attributes('aria-checked')).toBeFalsy()
	})

	it('selects locations of new value property', async () => {
		wrapper.setProps({ value: ['id0', 'id1'] })
		await Vue.nextTick()

		expect(locations.at(0).attributes('aria-checked')).toBeTruthy()
		expect(locations.at(1).attributes('aria-checked')).toBeTruthy()
		expect(locations.at(2).attributes('aria-checked')).toBeFalsy()
	})

	it('emits new selected locations when clicking on location', () => {
		locations.at(2).trigger('click')

		expect(wrapper.emitted().change[0][0]).toEqual(['id0', 'id2'])
	})

	it('emits new selected locations when pressing spacebar on location', () => {
		locations.at(2).trigger('keydown.space')

		expect(wrapper.emitted().change[0][0]).toEqual(['id0', 'id2'])
	})

	it('emits empty array when unselecting all locations', () => {
		locations.at(0).trigger('click')

		expect(wrapper.emitted().change[0][0]).toEqual([])
	})

	it('renders map with checkbox attributes', () => {
		expect(wrapper.element).toMatchSnapshot()
	})
})
