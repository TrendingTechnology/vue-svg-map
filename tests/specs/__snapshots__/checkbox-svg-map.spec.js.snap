// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxSvgMap component renders map with checkbox attributes 1`] = `
<svg
  aria-label="label"
  class="svg-map"
  map="[object Object]"
  role="group"
  viewBox="viewBox"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    aria-checked="true"
    aria-label="name0"
    class="svg-map__location"
    d="path0"
    id="id0"
    name="name0"
    role="checkbox"
    tabindex="0"
  />
  <path
    aria-label="name1"
    class="svg-map__location"
    d="path1"
    id="id1"
    name="name1"
    role="checkbox"
    tabindex="0"
  />
  <path
    aria-label="name2"
    class="svg-map__location"
    d="path2"
    id="id2"
    name="name2"
    role="checkbox"
    tabindex="0"
  />
</svg>
`;
